test/pool.test.ts

```diff
@@ -1,40 +1,85 @@
 import {PoolFactory, ReceiverWeightsTestFactory} from "../ethers-contracts";
 import buidler from "@nomiclabs/buidler";
 
-import {assert} from "chai";
+import {assert, expect} from "chai";
+
+async function addr(idx: number) {
+  return await (await signer(idx)).getAddress();
+}
+
+async function signer(idx: number) {
+  return (await buidler.ethers.getSigners())[idx];
+}
+
+const CYCLE_BLOCKS = 10;
 
 async function deployPool() {
-  const [signer] = await buidler.ethers.getSigners();
-  return new PoolFactory(signer).deploy().then((pool) => pool.deployed());
+  return new PoolFactory(await signer(0))
+    .deploy(CYCLE_BLOCKS)
+    .then((pool) => pool.deployed());
 }
 
-describe("Pool", function () {
-  it("rejects withdrawal from an empty account", async function () {
-    const pool = await deployPool();
-    await pool
-      .withdraw(1)
-      .then(() => assert.fail())
-      .catch((error) =>
-        assert(error.message.endsWith("Not enough funds in account"))
-      );
-  });
+async function mineBlocksUntilCycleEnd() {
+  const blockNumber = await buidler.ethers.provider.getBlockNumber();
+  await mineBlocks(CYCLE_BLOCKS - (blockNumber % CYCLE_BLOCKS));
+}
 
-  it("allows withdrawal from an account up to its value", async function () {
-    const pool = await deployPool();
-    await pool.topUp({value: 100});
-    await pool.withdraw(99);
-    await pool.withdraw(1);
-  });
+async function mineBlocks(count: number) {
+  for (let i = 0; i < count; i++) {
+    await buidler.ethers.provider.send("evm_mine", []);
+  }
+}
 
-  it("rejects withdrawal from an account over its value", async function () {
+describe("Pool", function () {
+  it("Sends some funds between accounts", async function () {
     const pool = await deployPool();
+    await mineBlocksUntilCycleEnd();
```

Comment:
The cause of this is the following: If `block.number < cycleBlocks` then we set `receiver.lastCollectedCycle = 0` in `setDeltasFromNow` which coincides with the uninitialized value. This means that `collect()` will return early and not collect anything. I think we can ignore this edge but we should document it properly and also include the fix (i.e. advancing the block number) in `deployPool` so that we avoid falling into that pitfall again.

Reply:
Good catch, it's a bug! But it's not that critical, it's a problem that if we start sending funds in the first `cycleBlocks` of the blockchain, the receiver is going to be initialized with `lastCollectedCycle = 0`. It's indistinguishable from being uninitialized at all and makes `collect` return early.

I can't see an easy and elegant way to add support for blocks before `cycleBlocks`. I don't think that it's worth the effort too, Ethereum is never going back to 0. I'll document it and leave it as it is.

Reply:
I think we have a race condition here ;). As mentioned in my previous comment I’d be ok with not fixing it since we’ll never run into this issue.

Reply:
Sorry, GitHub may be acting up again :( Not fixing it was my initial thought too, but if it's not much effort, it's better to have a sound implementation and not having to document and work around the issues.

Reply:
Fixed, it actually made code a little easier to understand

Reply:
I guess we can remove this call then, right?